base/compiler/optimize.jl

```diff
@@ -772,9 +772,9 @@ function inline_worthy(ir::IRCode,
         stmt = ir.stmts[line][:inst]
         thiscost = statement_or_branch_cost(stmt, line, ir, ir.sptypes, union_penalties, params)
         bodycost = plus_saturate(bodycost, thiscost)
-        bodycost > cost_threshold && return false
+        bodycost > cost_threshold && return 0
```

Comment:
`0` seems like the wrong value here, since the value is a cost. `typemax(UInt16)`?

Reply:
similarly elsewhere in this PR

Reply:
Good call, I think this is what was causing a lot of problems with things being improperly inlined. Building at normal speeds and passing nearly every relevant test now